### PR TITLE
Some test code cleanup

### PR DESCRIPTION
**Pull request by camsteffen**
changelog: none

Mainly moves /clippy_workspace_tests into /tests and combines the two dogfood tests which can't run concurrently.

### CODE REVIEW

File: tests/dogfood.rs [hunk 1]
```diff
@@ -3,179 +3,26 @@
...
+    if IS_RUSTC_TEST_SUITE {
         return;
     }
 
-    // NOTE: `path_dep` crate is omitted on purpose here
-    for project in &[
-        "clippy_workspace_tests",
-        "clippy_workspace_tests/src",
-        "clippy_workspace_tests/subcrate",
-        "clippy_workspace_tests/subcrate/src",
```
Comment (camsteffen) [inline tests/dogfood.rs:167]: These aren't dogfooded anymore which is less testing strictly speaking. But we're already running clippy against these in the actual workspace test.

File: .github/workflows/clippy_bors.yml [hunk 2]
```diff
@@ -133,14 +133,6 @@ jobs:
...
       working-directory: clippy_dev
 
-    - name: Test cargo-clippy
-      run: ../target/debug/cargo-clippy
-      working-directory: clippy_workspace_tests
-
-    - name: Test cargo-clippy --fix
-      run: ../target/debug/cargo-clippy clippy --fix
-      working-directory: clippy_workspace_tests
-
```
Comment (camsteffen) [inline .github/workflows/clippy_bors.yml:143]: These could be replaced with the new directory. I'm just not sure what this is supposed to test, and the test should probably be contained in the new `workspace.rs`.

Comment (flip1995) [reply .github/workflows/clippy_bors.yml:143]: Yeah, no idea why this is tested here.

File: tests/dogfood.rs [hunk 3]
```diff
@@ -3,179 +3,26 @@
 //!
 //! See [Eating your own dog food](https://en.wikipedia.org/wiki/Eating_your_own_dog_food) for context
 
-// Dogfood cannot run on Windows
-#![cfg(not(windows))]
```
Comment (flip1995) [inline tests/dogfood.rs:7]: The windows and mac runners are our slowest runners in CI and also the current bottleneck. This added another 1-2min to the windows runner. So I wonder if we should only run those tests on `unix`. An  argument against that is, that contributors using windows/mac, won't see dogfood errors until they submit a PR.

**Review by flip1995 — APPROVED** [review]
LGTM. One more question about dogfood on windows. If you think it should be enabled feel free to r=me.

File: tests/dogfood.rs (hunk 3 above)
Comment (camsteffen) [reply tests/dogfood.rs:7]: Thanks for reviewing this.

Maybe I can find a way to get the best of both worlds using a flag or environment variable.

**Comment by flip1995** [comment]
That's a neat solution! Thanks!

**Pull request merged**